internal/commands/server/relocate.go

```diff
@@ -0,0 +1,62 @@
+package server
+
+import (
+	"fmt"
+
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/commands"
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/completion"
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/output"
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/resolver"
+	"github.com/spf13/pflag"
+
+	"github.com/UpCloudLtd/upcloud-go-api/v8/upcloud/request"
+)
+
+// RelocateCommand creates the "server relocate" command
+func RelocateCommand() commands.Command {
+	return &relocateCommand{
+		BaseCommand: commands.New(
+			"relocate",
+			"Relocate a server",
+			"upctl server relocate 00038afc-d526-4148-af0e-d2f1eeaded9b --zone fi-priv-example",
+			"upctl server relocate 00038afc-d526-4148-af0e-d2f1eeaded9b 0053a6f5-e6d1-4b0b-b9dc-b90d0894e8d0 --zone fi-priv-example",
+			"upctl server relocate my_server1 --zone fi-priv-example",
+		),
+	}
+}
+
+type relocateCommand struct {
+	*commands.BaseCommand
+	completion.StoppedServer
+	resolver.CachingServer
+	zone string
+}
+
+// InitCommand implements Command.InitCommand
+func (s *relocateCommand) InitCommand() {
+	fs := &pflag.FlagSet{}
```

Comment:
Could add a long description here that briefly explains when relocating a server is possible. See [server create](https://github.com/UpCloudLtd/upcloud-cli/blob/main/internal/commands/server/create.go#L260) for example.